src/state.rs

```diff
@@ -43,6 +40,27 @@ pub struct StateDiff {
     pub nonces: IndexMap<ContractAddress, Nonce>,
     pub replaced_classes: IndexMap<ContractAddress, ClassHash>,
 }
+// TODO find a smarter way than using JSON
+// Start refactoring with `Program` struct and then `DeprecatedContractClass`
+impl Encode for StateDiff {
+    fn encode(&self) -> Vec<u8> {
+        let json_repr: String = json!(self).to_string();
+        json_repr.encode()
+    }
+}
+
+// TODO find a smarter way than using JSON
+// Start refactoring with `Program` struct and then `DeprecatedContractClass`
+impl Decode for StateDiff {
+    fn decode<I: parity_scale_codec::Input>(
+        input: &mut I,
+    ) -> Result<Self, parity_scale_codec::Error> {
+        let json_repr = <String>::decode(input)?;
+        serde_json::from_str(&json_repr).map_err(|_e| {
+            parity_scale_codec::Error::from("serde_json deserialization error for ContractClass")
+        })
+    }
+}
```

Comment:
Same thing:
- you don't have to go through json here, you can just encode each field successively
- add test